test: disable commitlog O_DSYNC, preallocation

avikivity opened the pull request:
Commitlog O_DSYNC is intended to make Raft and schema writes durable in the face of power loss. To make O_DSYNC performant, we preallocate the commitlog segments, so that the commitlog writes only change file data and not file metadata (which would require the filesystem to commit its own log).

However, in tests, this causes each ScyllaDB instance to write 384MB of commitlog segments. This overloads the disks and slows everything down.

Fix this by disabling O_DSYNC (and therefore preallocation) during the tests. They can't survive power loss, and run with --unsafe-bypass-fsync anyway.

nyh commented:
Looks good, I'll wait for the CI before merging.

I'm curious what kind of test speedup you noticed in what situation. I see test/cql-pytest/run's single-node boot time on ram disk go down from around 0.35 seconds to 0.3 seconds, which is nice, but I wonder where you saw a more significant impact - is it when running on some sort of slow disks?

avikivity commented:
> Looks good, I'll wait for the CI before merging.
> 
> I'm curious what kind of test speedup you noticed in what situation. I see test/cql-pytest/run's single-node boot time on ram disk go down from around 0.35 seconds to 0.3 seconds, which is nice, but I wonder where you saw a more significant impact - is it when running on some sort of slow disks?

I didn't test on slow disks. I think that an individual test won't be sped up by much - the writes go to page cache and are flushed in the background. But when you have many such tests, the kernel is eventually forced to flush and you get contention.

scylladb-promoter commented:
CI state **`ABORTED`** - https://jenkins.scylladb.com/job/releng/job/Scylla-CI/3874/

nyh commented:
@avikivity tests failed with:
```
[2023-01-17T21:15:17.928Z] error: unrecognised option '--commitlog-use-o-dsync'
```

avikivity commented:
Update:
 - moved unit test changes from command line (where only seastar options are recognized) to cql_test_env
 - rebased
 - tested with entire test suite

scylladb-promoter commented:
CI state **`SUCCESS`** - https://jenkins.scylladb.com/job/releng/job/Scylla-CI/3886/

avikivity commented:
Success!

kbr-scylla commented:
Thanks, queued.

We might want to backport this to branch-5.2 once it's merged, we're running out of disk space on arm machines during tests :)

kbr-scylla commented:
cc @kostja
